cmd/guard.go

```diff
@@ -0,0 +1,83 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/leancodepl/poe2arb/flutter"
+	"github.com/spf13/cobra"
+)
+
+type flutterConfigKey int
+
+const key flutterConfigKey = 1
+
+func contextWithFlutterConfig(ctx context.Context, flutterConfig *flutter.FlutterConfig) context.Context {
+	return context.WithValue(ctx, key, flutterConfig)
+}
+
+func flutterConfigFromContext(ctx context.Context) (*flutter.FlutterConfig, bool) {
+	flutterCfg, ok := ctx.Value(key).(*flutter.FlutterConfig)
+	return flutterCfg, ok
+}
+
+func ensureSufficientVersion(versionConstraint string) error {
+	if versionConstraint == "" {
+		return nil
+	}
+
+	constraint, err := newConstraintFromString(versionConstraint)
+	if err != nil {
+		return fmt.Errorf("invalid poe2arb-version format in l10n.yaml: %s", versionConstraint)
+	}
+
+	version, _ := version.NewVersion(Version)
+
+	if !constraint.Check(version) {
+		return fmt.Errorf("Poe2Arb version %s does not match constraint %s defined in l10n.yaml", version, versionConstraint)
+	}
+
+	return nil
+}
+
+func newConstraintFromString(versionConstraint string) (version.Constraints, error) {
+	return version.NewConstraint(strings.ReplaceAll(versionConstraint, " ", ", "))
```

Comment:
If the syntax doesn't use commas we can just use the library's syntax, I wouldn't alter it. Then this function won't be needed